src/v/cluster_link/replication/types.h

```diff
@@ -15,11 +15,30 @@
 #include "model/record.h"
 #include "ssx/semaphore.h"
 
+#include <cstdio>
+
 namespace cluster_link::replication {
 
 struct fetch_data {
     chunked_vector<::model::record_batch> batches;
     ssx::semaphore_units units;
 };
 
+struct fetch_counters {
+    size_t n_bytes{};
+    size_t n_records{};
+
+    static fetch_counters from_fetch_data(const fetch_data& data);
+
+    fetch_counters& operator+=(const fetch_counters& rhs) {
+        n_bytes += rhs.n_bytes;
+        n_records += rhs.n_records;
+        return *this;
+    }
+};
+
+struct probe_data {
```

Comment:
naming nit:  shoudln't this read "data_probe" instead of "probe_data", the latter sounds like a verb